cmake/system_check.cmake

```diff
@@ -109,7 +109,7 @@ else()
 endif ()
 
 if (NOT BINARY)
-  if (X86_64 OR ARM64 OR POWER OR MIPS64 OR LOONGARCH64 OR RISCV64)
+  if (X86_64 OR ARM64 OR MIPS64 OR LOONGARCH64 OR RISCV64 OR (POWER AND NOT (CMAKE_OSX_ARCHITECTURES STREQUAL "ppc")))
```

Comment:
I wonder if the CMAKE_SIZEOF_VOID_P trick would work on power, to define BINARY right when power is detected instead of adding such an involved construct here ?

Reply:
Do you know what exactly it does? Here is the issue:

1. Physical PowerPC cpus may or may not support 64-bit binaries (G4 does not, G5 does). On G5 we may build for `ppc` or `ppc64`.
2. Kernel is always 32-bit on macOS ppc.
3. `uname` output will be `powerpc` or `Power Macintosh`, so it is useless to differentiate between `ppc` and `ppc64`.
4. On Rosetta physical cpu is Intel (32- or 64-bit), while building is for `ppc` (only 32-bit, no support for `ppc64` in Rosetta).

Reply:
Was not aware of this, that's why I'm asking - as far as I know, the powermac platform has been obsolete for more than 15 years, and cmake support in OpenBLAS is more recent - no wonder this was overlooked for so long

Reply:
I think using CMAKE_OSX_ARCHITECTURES is the right thing to go with – it works reliably both natively and in Rosetta. (Macports uses that to set architecture as well.)
Unless there are some specific concerns about it.

Reply:
Alright, thanks.